### PR TITLE
Add container ECS fields in kubernetes metadata

### PR DESCRIPTION
**Pull request by ChrsMark**
## What does this PR do?
This PR adds `container.id`, `container.runtime` and `container.image.name` in kubernetes metadata to be compliant with https://www.elastic.co/guide/en/ecs/current/ecs-container.html

## Why is it important?
In final events the metadata do not include `container.id`, `container.runtime` and `container.image.name`. 

## Related issues

- Closes https://github.com/elastic/beats/issues/20982
- Related https://github.com/elastic/beats/issues/10926

## How to test this PR
1. Configure and run Filebeat in k8s:
```
filebeat.autodiscover:
      providers:
        - type: kubernetes
          node: ${NODE_NAME}
          hints.enabled: true
          hints.default_config:
            type: container
            paths:
              - /var/log/containers/*${data.kubernetes.container.id}.log
```
Make sure that `container.id`, `container.runtime` and `container.image.name` exist in the final documents in ES.

2. Configure and run Filebeat in k8s:
```
filebeat.inputs:
    - type: container
      paths:
        - /var/log/containers/*.log
      processors:
        - add_kubernetes_metadata:
            host: ${NODE_NAME}
            indexers:
            - container:
            matchers:
            - logs_path:
                logs_path: "/var/log/containers/"
```

## Screenshots
Screenshot 2020-09-08 at 13 56 03

### CODE REVIEW

File: libbeat/autodiscover/providers/kubernetes/pod.go [hunk 1]
```diff
@@ -334,19 +334,17 @@ func (p *pod) emitEvents(pod *kubernetes.Pod, flag string, containers []kubernet
...
-		}
-		meta := p.metagen.Generate(pod, metadata.WithFields("container.name", c.Name),
-			metadata.WithFields("container.image", c.Image))
+		
+		meta := p.metagen.Generate(
+			pod,
+			metadata.WithFields("container.name", c.Name),
+			metadata.WithFields("container.image", c.Image),
+			metadata.WithFields("container.id", cid),
+			metadata.WithFields("container.runtime", runtimes[c.Name]),
```
Comment (jsoriano) [inline libbeat/autodiscover/providers/kubernetes/pod.go:343]: I think `kubernetes.container.id` and `kubernetes.container.runtime` don't have a mapping, could you confirm and add it?

I think kubernetes fields are defined in `libbeat/processors/add_kubernetes_metadata/_meta/fields.yml`.

File: libbeat/autodiscover/providers/kubernetes/pod.go [hunk 2]
```diff
@@ -334,19 +334,17 @@ func (p *pod) emitEvents(pod *kubernetes.Pod, flag string, containers []kubernet
 		// This must be an id that doesn't depend on the state of the container
 		// so it works also on `stop` if containers have been already deleted.
 		eventID := fmt.Sprintf("%s.%s", pod.GetObjectMeta().GetUID(), c.Name)
-
-		cmeta := common.MapStr{
-			"id":      cid,
-			"name":    c.Name,
-			"image":   c.Image,
-			"runtime": runtimes[c.Name],
-		}
```
Comment (jsoriano) [inline libbeat/autodiscover/providers/kubernetes/pod.go:343]: :+1:

**Review by jsoriano — APPROVED** [review]
Good catch! LGTM

It'd be nice to have some test to detect the loss of these fields. We would also need to check if `add_kubernetes_metadata` has similar issues.

File: libbeat/autodiscover/providers/kubernetes/pod.go (hunk 1 above)
Comment (ChrsMark) [reply libbeat/autodiscover/providers/kubernetes/pod.go:343]: added

File: libbeat/processors/add_kubernetes_metadata/indexers.go [hunk 3]
```diff
@@ -234,14 +239,29 @@ func (h *IPPortIndexer) GetMetadata(pod *kubernetes.Pod) []MetadataIndex {
...
 	for _, container := range pod.Spec.Containers {
 		for _, port := range container.Ports {
 			if port.ContainerPort != 0 {
 
 				m = append(m, MetadataIndex{
 					Index: fmt.Sprintf("%s:%d", pod.Status.PodIP, port.ContainerPort),
-					Data: h.metaGen.Generate(pod, metadata.WithFields("container.name", container.Name),
-						metadata.WithFields("container.image", container.Image)),
+					Data: h.metaGen.Generate(
+						pod, metadata.WithFields("container.name", container.Name),
```
Comment (jsoriano) [inline libbeat/processors/add_kubernetes_metadata/indexers.go:260]: Nit.
```suggestion
						pod,
						metadata.WithFields("container.name", container.Name),
```

**Review by jsoriano — APPROVED** [review]
Thanks!

**Pull request merged**